### PR TITLE
Add customizable endpoint exclusion

### PR DESCRIPTION
**Pull request by ahauschulte**
Camel-plantuml is a powerful tool for visualizing Camel routes, from small and simple routes to larger and more complex ones. In my use cases with complex routes, however, I found that certain endpoints were irrelevant to the specific diagrams I wanted to create, and their inclusion sometimes reduced clarity.

This PR introduces a new URL parameter (`uriFilterPattern`), allowing users to filter out specified endpoints in the PlantUML output, thereby better aligning the visual output to specific needs and enhancing readability for focused use cases. I’ve included two unit tests to demonstrate the parameter’s use.

### CODE REVIEW

**Comment by ncasaux** [comment]
Thank you @ahauschulte for your interest in this project and for your suggestion! 
I could not find the unit tests you mentioned, or maybe I missed something ?

**Comment by ahauschulte** [comment]
Thank you for pointing out that the tests were missing. I've updated the pull request.

File: camel-plantuml-jar/src/main/java/io/github/ncasaux/camelplantuml/generator/RoutesDiagramGenerator.java [hunk 1]
```diff
@@ -47,6 +48,11 @@ public static String generateUmlString(HashMap<String, RouteInfo> routesInfo) th
                 }
             }
 
+            if (parameters.uriFilterPattern().matcher(routeEndpointBaseUri).matches()) {
```
Comment (ncasaux) [inline camel-plantuml-jar/src/main/java/io/github/ncasaux/camelplantuml/generator/RoutesDiagramGenerator.java:51]: If we want to filter based on the route Id, I think we should not use the `routeEndpointBaseUri` but the `routeId` instead.

**Review by ncasaux — CHANGES_REQUESTED** [review]
I think there is a small bug where I added a comment.
Anyway, you really perfectly integrated your feature into my code, congratulations ! 👍

**Comment by ahauschulte** [comment]
Thank you for the feedback and your kind words.

I reviewed the filter section in the `RoutesDiagramGenerator` class and think that filtering should indeed be based on the `routeEndPointBaseUri`. Filtering by `routeId` would not work as intended (the test in `DefaultExampleWithUriFilterPatternTest.java` would fail in that case and the generated diagram would be incorrect).

Perhaps the log statement within the `if` block was unclear? I’ve made an effort to clarify it.

**Comment by ncasaux** [comment]
Your modification makes sense, as it's a URI filter, not an ID filter.
Thanks for adding the routeEndPointBaseUri in the log.

**Comment by ncasaux** [comment]
I've published version 1.8.0, it should be available soon in Maven Central.
Thanks again for your PR! 👍

**Comment by ahauschulte** [comment]
Thanks for merging the PR; and thank you for camel-plantuml, it's a very useful tool.

**Pull request merged**